src/cli/onefuzz/debug.py

```diff
@@ -262,6 +266,88 @@ def download_files(self, job_id: UUID_EXPANSION, output: Directory) -> None:
             subprocess.check_output([azcopy, "sync", to_download[name], outdir])
 
 
+class DebugLog(Command):
+    def _convert(self, raw_data: Any) -> Dict[str, List[Dict[str, Any]]]:
+        results = {}
+        for table in raw_data.tables:
+            result = []
+            for row in table.rows:
+                converted = {
+                    table.columns[x].name: y
+                    for (x, y) in enumerate(row)
+                    if y not in [None, ""]
+                }
+                if "customDimensions" in converted:
+                    converted["customDimensions"] = json.loads(
+                        converted["customDimensions"]
+                    )
+                result.append(converted)
+            results[table.name] = result
+        return results
+
+    def query(
+        self, log_query: str, *, timespan: str = "PT24H", raw: bool = False
+    ) -> Any:
+        """
+        Perform an Application Insights query
+
+        :param str log_query: Query to send to Application Insights
+        :param str timespan: ISO 8601 duration format
+        :param bool raw: Do not simplify the data result
+        """
+        creds, _ = get_azure_cli_credentials(
+            resource="https://api.applicationinsights.io"
+        )
+        client = ApplicationInsightsDataClient(creds)
+
+        app_id = self.onefuzz.info.get().insights_appid
+        if app_id is None:
+            raise Exception("instance does not have an insights_appid")
+        raw_data = client.query.execute(
+            app_id, body=QueryBody(query=log_query, timespan=timespan)
+        )
+        if "error" in raw_data.additional_properties:
+            raise Exception(
+                "Error performing query: %s" % raw_data.additional_properties["error"]
+            )
+        if raw:
+            return raw_data
+        return self._convert(raw_data)
+
+    def keyword(
+        self,
+        value: str,
+        *,
+        timespan: str = "PT24H",
+        limit: Optional[int] = None,
+        raw: bool = False,
+    ) -> Any:
+        """
+        Perform an Application Insights keyword query akin to "Transaction Search"
+
+        :param str value: Keyword to query Application Insights
+        :param str timespan: ISO 8601 duration format
+        :param int limit: Limit the number of records returned
+        :param bool raw: Do not simplify the data result
+        """
+        components = ["union isfuzzy=true exceptions, traces, customEvents"]
```

Comment:
Great, I was hoping this is how you'd approach the query building.

Reply:
I suggest documenting that you are building KQL, which should help make it clearer what `|` means below, for example.